SECURITY.md

```diff
@@ -30,9 +52,11 @@ Extensions are sandboxed from the host system by operating via APIs.  This secur
 
 ## Communications security (COMSEC)
 
-Plugins are loaded into the same java virtual machine instance allowing communicate to OpenSearch through in process java APIs.  Plugins can issue API requests to the OpenSearch hosts reusing the standard node-to-node communications, internally called the transport client.
+Plugins are loaded into the same java virtual machine instance allowing communication to OpenSearch through in-process java APIs, although this behavior will be deprecated over time in favor of using a common API endpoint.  Plugins can issue API requests to the OpenSearch hosts reusing the standard node-to-node communications, internally called the transport client.
+
+Extensions of OpenSearch communicate via https requests between the nodes on the cluster and the extensions endpoint(s).  This is a bi-directional communication and also allows extensions to contact the OpenSearch cluster through its available APIs.
```

Comment:
~So are we agreeing upon having HTTPs for extension <-> OpenSearch communication ? 
The design built today is via transport/TCP.~
I see the scary warning at bottom :)